src/neural-network.js

```diff
@@ -830,6 +830,9 @@ export default class NeuralNetwork {
       }
     }
     if (json.hasOwnProperty('trainOpts')) {
+      if (!json.trainOpts.timeout) {
+        json.trainOpts.timeout = Infinity;
```

Comment:
The problem with the approach is that it modifies the object being sent into brain. This is where I would have went as well, but it may be more advantagoes to do something like this, so as to ensure we aren't changing any values:

```js
if (json.hasOwnProperty('trainOpts')) {
  this._updateTrainingOptions({
    ...json.trainOpts,
    timeout: json.trainOpts.timeout === null ? Infinity : json.trainOpts.timeout
  });
}
```

This allows us to handle the exact situation where timeout is parsed to null, and treated as infinity, but has no side effects of altering the object the is sent to us.